### ISSUE TITLE
blur keeps wrong options around

### ISSUE THREAD

**Issue by KristerV** [maxmarcon/live_select#29]
# How to reproduce

1. write something in the live_select (see how options change)
2. click outside of the input (options dropdown dismissed, input cleared)
3. click on the input again
4. notice how the old options are shown even though input is empty

# What should happen

either the initial options should be restored or a possibility to detect the blur event so i can reset manually.

---

i'll make a minimal example if required. thought i'd pick your brain first.

**Comment by maxmarcon** [comment]
yeah it's one of those things with no clear right or wrong IMO: some people will like the old selection to stay, some others won't. 

Don't recall exactly why I didn't clear the list of options on blur - maybe it just felt better that way to me.

Question is how to make this customizable behavior. Simpler thing that comes to mind would be to add a `clear_options_on_blur` assign or exposing a `blur` event.

"Problem" with the `blur` event is that all events sent to the parent have to go through the client (but it already happens for the `live_select_change` event, so it's probably not a big deal).

**Comment by KristerV** [comment]
current problem is that there is no way to choose what to display. can't reset, can't empty. UI is basically in a stuck state where user has to click something else first, then option comes back.

i think just implementing the blur event without changing the current behaviour is fine.

**Comment by maxmarcon** [comment]
I added 2 new assigns:

* `phx-focus`: focus event, will be sent to the parent when the input gets focus
* `phx-blur`: blur event, will be sent to the parent when the input loses focus. This includes the case where the user selects an entry from the dropdown in single mode.

Both events will receive the `live_select` component's `id` in the payload.

`live_select`'s behavior you were describing above is consistent, in the sense that live_select will _never_ change the `options`: It's up to the user to update them as a response to an event.

So now you can react to the above events. When the input is blurred you can clear the list of options, and when it gets focus again you can restore your set of default options. Like this:

```elixir
<.live_select phx-focus="focus" phx-blur="blur" options={@default_options} />
```

```elixir
def handle_event("focus", %{id: live_select_id}, socket) do 
 send_update(LiveSelect.Component, id: live_select_id, options: @default_options)

 {:noreply, socket}
end

def handle_event("blur", %{id: live_select_id}, socket) do 
 send_update(LiveSelect.Component, id: live_select_id, options: [])

 {:noreply, socket}
end

```

@KristerV  Does it make sense? Could you please try this out on master and let me know? Thanks!!!! 🙏

**Comment by maxmarcon** [comment]
Closed with 9c4a6517e8276aa3121207e950c735c78b5181a0
